backend/LeetWars.Core/LeetWars.Core.BLL/Services/UserService.cs

```diff
@@ -52,7 +56,17 @@ public async Task<UserDto> CreateUserAsync(NewUserDto userDto)
             throw new InvalidOperationException($"A user with email {userDto.Email} is already registered.");
         }
 
+        bool isExistingUserName = await CheckIsExistingUserNameAsync(userDto.UserName);
+
+        if (isExistingUserName)
+        {
+            userDto.UserName = await GenerateUniqueUsername(userDto.Email);
+        }
+
         var newUser = _mapper.Map<NewUserDto, User>(userDto);
+
+        newUser.RegisteredAt = DateTime.UtcNow;

```

Comment:
maybe it would be better to add a field "RegistredAt" to NewUserDto and set DateTime.UtcNow in dto?

Reply:
I think it would be great to set this property by the database, but it requires from me migration, so for now I decided to leave like this

Reply:
or set in UserEntity

Reply:
ok